perf/sawtooth_perf/src/batch_submit.rs

```diff
@@ -154,19 +167,388 @@ pub fn http_submitter(target: String, rate: u64, receiver: Arc<Mutex<mpsc::Recei
+                                            }
+                                            return Err(WorkloadError::UnknownRestApiError)
+                                    },
+                                    Err(err) => {
+                                        println!("Error converting UTF8: {}", err);
+                                        return Err(WorkloadError::Utf8Error(err))
+                                    },
+                                }
+                            },
+
+                            None => Err(WorkloadError::UnknownRestApiError),
+                    }})
+                    .and_then(move |uri: Result<Uri, WorkloadError> | {
+                        match uri {
+                            Ok(uri) => {
+                                let get_status = Client::new(&handle_clone_2).get(uri)
+                                    .and_then(|response| response.body().concat2())
+                                    .map(|body: Chunk| {
+                                        let batch_statuses_response = serde_json::from_str(from_utf8(&body.to_vec())?)?;
+                                        Ok(batch_statuses_response)
+                                    })
+                                    .map(move |batch_statuses_response: Result<BatchStatusesResponse, WorkloadError>| {
+                                        let counter_ccc = counter_cc.clone();
+                                        let counter_iii = counter_ii.clone();
+                                        match batch_statuses_response?.data.get(0) {
+                                            Some(batch_status) => {
+                                                if batch_status.status == "COMMITTED" {
+                                                    counter_ccc.increment_committed();
+                                                } else if batch_status.status == "INVALID" {
+                                                    counter_iii.increment_invalid();
+                                                }
+                                            },
+                                            None => println!("Batch status came back with no status."),
+                                        };
+                                        Ok(())
+                                    }).and_then(|_: Result<(), WorkloadError>| future::ok(()) ).map_err(|_| ());
+
+                                    Ok(handle_clone_2.spawn(get_status))
+                            },
+                            Err(_) => {Ok(())},
+                        }
+                    }).and_then(|_| future::ok(()) ).map_err(|_| () );
+                Ok(handle_clone_3.spawn(work))
+            },
+            
+            Err(err) => Err(err),
+            }
+            }).for_each(|_| future::ok(())).map_err(|err| WorkloadError::from(err));
+
+    core.run(stream_work)
+}
```

Comment:
Some of the above closures should be moved to first class functions if possible so it is easier to understand what is happening above. The above is very hard to make sense of. It may even be a good idea to refactor the above into a struct/impl and move it out of this file.

Reply:
This could be done in a separate PR - it doesn't necessarily gate the functionality.

Reply:
https://jira.hyperledger.org/browse/STL-1014 I opened a Jira story to do the refactor.